### PR TITLE
board, nucleo144-f446: update spi frequency config

### PR DESCRIPTION
**Pull request by smlng**
@MichelRottleuthner was investigating some SPI problems on this board, I fiddled around with that a bit and updated the spi_divtable table. 

Hence, @MichelRottleuthner please try if this helps ...

### CODE REVIEW

**Comment by smlng** [comment]
btw. @MichelRottleuthner did you this #7123 and that #7158, might be related?

**Comment by MichelRottleuthner** [comment]
Thanks for keeping track of that! 
TL;DR: it helps a bit ;)

Resulting measures with periph_spi test when sending 'abc' with your changes applied:

1)`init 0 0 0 3 14`
clock should be 100 kHz, real: ~350 kHz (vs. ~330 kHz before)
clock state is NOT reset to zero properly after transmission.

2)`init 0 0 1 3 14`
clock should be 400 kHz, real: ~350 kHz (vs. ~330 kHz before)
clock state is NOT reset to zero properly after transmission.

3)`init 0 0 2 3 14`
clock should be 1 MHz, real: ~700 kHz (vs. ~330 kHz before)
clock state is NOT reset to zero properly after transmission.

4)`init 0 0 3 3 14`
clock should be 5 MHz, real: ~5 MHz (vs. ~2.6 MHz before)
clock state is reset to zero properly after transmission.

5)`init 0 0 4 3 14`
clock should be 10 Mhz -> real:  ~8-12 MHz (vs. ~5 MHz before)
clock state is reset to zero properly after transmission.

The last two seem OK, just keep in mind I used a 24MS logic analyzer.

To illustrate the Problem with the clock state not returning see the following screenshots.
A) not returning to zero, using config (1) from above:
spi_0
B) returning to zero, using config (5) from above:
spi_4

If you start another transmission after (A), you get this:
spi_0_1

**Comment by MichelRottleuthner** [comment]
>btw. @MichelRottleuthner did you this #7123 and that #7158, might be related?

Not yet, but thats indeed something to look at.

**Comment by smlng** [comment]
you could also try to revert #7158 and try again

**Comment by MichelRottleuthner** [comment]
Tested, but seems like that isn't the cause. In the current master the spi_divtable is definitely strange as it says [`APB1 @90000000Hz`](https://github.com/RIOT-OS/RIOT/blame/master/boards/nucleo144-f446/include/periph_conf.h#L166) and [`APB2 @180000000Hz`](https://github.com/RIOT-OS/RIOT/blame/master/boards/nucleo144-f446/include/periph_conf.h#L173) but according to the [data sheet](http://www.st.com/content/ccc/resource/technical/document/datasheet/65/cb/75/50/53/d6/48/24/DM00141306.pdf/files/DM00141306.pdf/jcr:content/translations/en.DM00141306.pdf) P.21-3.13 that isn't even possible.

**Comment by vincent-d** [comment]
Actually, I changed the core clock speed in some nucleo in #7158 and `spi_divtable` have not been updated.

It would explain the clock speed differences, but not the clock level issues.

File: boards/nucleo144-f446/include/periph_conf.h [hunk 1]
```diff
@@ -29,6 +29,7 @@ extern "C" {
  * @name    Clock system configuration
  * @{
  */
+#define P                       (4U)
```
Comment (haukepetersen) [inline boards/nucleo144-f446/include/periph_conf.h]: what is this about?

Comment (smlng) [inline boards/nucleo144-f446/include/periph_conf.h]: its the _PLL Output divisor_  see [here](https://github.com/RIOT-OS/RIOT/blob/master/cpu/stm32f4/stmclk.c#L50) and needs to be set to `4` to work with the cpu frequency here.

Comment (haukepetersen) [reply boards/nucleo144-f446/include/periph_conf.h]: I see. A comment on this line would not hurt... But I think this will be obsolete through #7477 in the near future anyway, right? Still, I would prefer to move this line down to the end of the clock config block.

**Comment by smlng** [comment]
@haukepetersen yes I also think this one will be obsolete with #7477 - basically it was _just_ a WIP/devel branch to document and try things I discussed offline with @MichelRottleuthner.

Shall I close right away, or wait for #7477 to be merged?

**Comment by haukepetersen** [comment]
true. I think I would prefer to wait for #7477 (which should be close to being merged) would be the way to proceed. But I think we should keep this PR open until then and verify that #7477 does solve the SPI issue described above. Do you agree?

**Comment by smlng** [comment]
@haukepetersen with #7477, fine to close this one?

**Comment by smlng** [comment]
@MichelRottleuthner does #7477 work for you?

**Comment by MichelRottleuthner** [comment]
retested - spi still seems to be broken on this board.

**Comment by smlng** [comment]
with this PR and/or with #7477?

**Comment by MichelRottleuthner** [comment]
with #7477 (i.e. current master)

**Comment by tcschmidt** [comment]
How to proceed: Fix broken  #7477  (in master) or this PR?

**Comment by MichelRottleuthner** [comment]
Seems like the issue is still there and moreover it's also affecting other MCUs. I tested with nucleo-f446ze, nucleo-l073rz and nucleo-l476rg and all of them show the problem where clock is not returning to idle state (see pic. A above). The problem with the clock starting again before transmitting the next bytes (C) is only present on nucleo-f446ze and nucleo-l073rz (so far).

The root cause of this seems to be that the driver is disabling the SPI peripheral to early after the last bit was clocked out, after the transmission in [spi_release](https://github.com/RIOT-OS/RIOT/blob/1e3360e1bf5fd466b4076f1cd7d2512d8d6a4360/cpu/stm32_common/periph/spi.c#L142). If you insert a small timeout of ~half an SPI-clock period before the disable command, you won't see the faulty behaviour. As this hapens only with slow SPI-clock settings (clk setting <=1 in spi periph test) I assume it's some kind of race condition based on internal clock-synchronisation problems where the hardware doesn't reach the point where it switches the clock back to idle before the peripheral is disabled by the riot driver. I double checked that the driver uses the correct procedure for turning off SPI (waiting for TXE, BSY etc.) and I can't find a problem there.
Possible workarounds I identified so far: 
1) don't disable SPI -> solves A and C, possibly bad for low power and/or shared peripheral usage 
2) change the SPI prescalers to not support the slow clock speeds on affected MCUs -> avoids A and C but decreases functionality
3) move setting the MSTR bit from [spi_acquire](https://github.com/RIOT-OS/RIOT/blob/1e3360e1bf5fd466b4076f1cd7d2512d8d6a4360/cpu/stm32_common/periph/spi.c#L131) to [spi_transfer_bytes](https://github.com/RIOT-OS/RIOT/blob/1e3360e1bf5fd466b4076f1cd7d2512d8d6a4360/cpu/stm32_common/periph/spi.c#L254) -> solves only C, but with no additional clock cycles it is at least ensured to not transmit borked data. Leaving the clock on the active level might not be a good thing though.
4) poll clock gpio till returnig to idle level before [disabling SPI](https://github.com/RIOT-OS/RIOT/blob/1e3360e1bf5fd466b4076f1cd7d2512d8d6a4360/cpu/stm32_common/periph/spi.c#L254) -> solves A and C - but a bit hacky

@smlng @MrKevinWeiss @haukepetersen any thoughts on how to proceed here?

**Comment by MrKevinWeiss** [comment]
1. No vote for me, even though this is the way the reference implementation says, it could pose a SPI problem when working with the PM, unpleasant to debug.

2. I may be OK with this if every board would be tested but since it could be board and config dependant the problem may still persist in certain cases, cleanest but requires testing many boards.  Best if we are going to implement interrupts or dma that depend on the register changes.

3. Nope since the sck line is still not correct (even though in code you don't see it)

4. I think this would be the best if no special AF things are needed.  You can make a nice static inline function to make it look clean but ultimately this is what the hardware should perform.  The problem I have with this could be the fact that it requires polling and my not be good interrupts or dma.

**Comment by miri64** [comment]
This was ACK'd a while back. What is the status here (apart from a required rebase).

**Comment by MrKevinWeiss** [comment]
I think we were offered a number of solutions and nobody selected anything :confused:

**Comment by MichelRottleuthner** [comment]
The PR changes to the clock config are already in master thanks to the unified SPI divtable code. 
I propose to close this PR and continue with one of the following steps regarding the additional problems found with the clock not returning to idle state:
I) add memo label / open issue and fix later with help of SPI HIL testing
III) open a PR implementing one of the above fixes and manually test all the boards
Any opinions?

**Comment by PeterKietzmann** [comment]
@MichelRottleuthner that sounds reasonable. Keeping this outdated PR open only for discussion doesn't make sense to me. Would you open an issue please?

**Comment by MichelRottleuthner** [comment]
Closing because changes are not valid anymore and issue #11104 is now in place to keep track of the bugs discovered above.
